Core/TopBuilder.cpp

```diff
@@ -705,6 +715,12 @@ llvm::Value* Builder::createGEP(llvm::Value* gepValue, llvm::ArrayRef<llvm::Valu
 }
 
 // Comments in header
+llvm::Value* Builder::createBitCast(llvm::Value* castValue, llvm::Type* castType)
```

Comment:
Hi John,

You will notice I redid the PR without bitcast.

The reason for the original design of the PR was that I thought it was the
most elegant way to do what we wanted to do with the design at hand.

Under the current regime, "incomplete" collapsing returns a pointer to a
vector. One way to access a component with Load/Store is to convert that
pointer into a pointer to the component type with bitcast and index off the
the pointer, which is what I did. This was based on a recipe that Samsung
asked me to use to avoid indexing into vectors, which they claimed was
deprecated. Interestingly, LLVM 3.6 turns this sequence into a GEP that
indexes into the vector, which is what you originally generated and Samsung
was trying to avoid. So I am doubtful that bitcast is the answer to our
problem of dealing with the negative effects of component accesses, at
least with 3.6.

Greg

Greg Fischer

Project Manager and Senior Engineer

LunarG, Inc. - 3D Driver Innovations

greg@lunarg.com <alan@lunarg.com>

On Fri, Apr 22, 2016 at 10:19 AM, John Kessenich <notifications@github.com>
wrote:

> In Core/TopBuilder.cpp
> <https://github.com/LunarG/LunarGLASS/pull/15#discussion_r60765179>:
>
> > @@ -705,6 +715,12 @@ llvm::Value* Builder::createGEP(llvm::Value* gepValue, llvm::ArrayRef<llvm::Valu
> >  }
> >
> >  // Comments in header
> > +llvm::Value* Builder::createBitCast(llvm::Value* castValue, llvm::Type* castType)
>
> Hopefully, this will not exist for pointer-type changing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LunarG/LunarGLASS/pull/15/files/3c324682685112ce7643fce2272053fc4ba4cfd2#r60765179>
>